src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,49 @@
 package core.basesyntax;
 
-class Pair {
+public class Pair<T, E> {
+    private T first;
+    private E second;
 
+    private Pair(T first, E second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public T getFirst() {
+        return first;
+    }
+
+    public E getSecond() {
+        return second;
+    }
+
+    public static <T, E> Pair<T, E> of(T first, E second) {
+        return new Pair<>(first, second);
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if (getClass().equals(object.getClass())) {
+            Pair<?, ?> pair = (Pair<?, ?>) object;
```

Comment:
please see checklist - what does it say about wildcards?

Reply:
Solved, replaced with parametrised types.